services/QuillLMS/spec/workers/set_impact_metrics_worker_spec.rb

```diff
@@ -5,54 +5,56 @@
-    let!(:second_schools_user) { create(:schools_users, school: low_income_school, user: second_classroom.owner)}
-    let!(:archived_activity_session) { create(:activity_session, user: User.where(role: 'student').first, visible: false )}
-
-    before(:all) do
-      @number_of_sentences = ActivitySession.unscoped.where(state: 'finished').count.floor(-5) * 10
-      @number_of_students = ActivitySession.unscoped.where(state: 'finished').count('DISTINCT(user_id)')
-      teachers_query = User.select(:id)
-        .joins(:units, classroom_units: :activity_sessions)
-        .where("activity_sessions.state = 'finished'")
-        .group('users.id')
-        .having('count(activity_sessions) > ?', SetImpactMetricsWorker::ACTIVITY_SESSION_MINIMUM)
-      teacher_ids = teachers_query.to_a.map(&:id)
-      @number_of_teachers = teachers_query.to_a.count.floor(-2)
-      schools_query = School.joins(:schools_users).where(schools_users: {user_id: teacher_ids}).distinct
-      @number_of_schools = schools_query.to_a.count
-      low_income_schools_query = schools_query.where("free_lunches > ?", SetImpactMetricsWorker::FREE_LUNCH_MINIMUM)
-      @number_of_low_income_schools = low_income_schools_query.to_a.count
+  context '#perform' do
+    let(:activity_sessions) { create_list(:activity_session, 100)}
+
+    let(:example_json1) {
+      activity_sessions.map {|as| {"id" => as.id}}
+    }
+
+    let(:teachers) { create_list(:teacher, 150)}
+
+    let(:example_json2) {
+      teachers.map {|t| {"id" => t.id}}
+    }
+
+    let(:example_json3) {
+      [
+        {"id" => 333, "free_lunches" => 30},
+        {"id" => 334, "free_lunches" => 0},
+        {"id" => 545, "free_lunches" => 54}
+      ]
+    }
+
+    before do
+      allow(QuillBigQuery::ActivitiesAllTimeQuery).to receive(:run).and_return(example_json1)
+      allow(QuillBigQuery::ActiveTeachersAllTimeQuery).to receive(:run).and_return(example_json2)
+      allow(QuillBigQuery::SchoolsContainingCertainTeachersQuery).to receive(:run).with(teachers.map(&:id)).and_return(example_json3)
     end
 
     it 'should set the NUMBER_OF_SENTENCES redis value' do
       subject.perform
-      expect($redis.get(PagesController::NUMBER_OF_SENTENCES)).to eq(@number_of_sentences.to_s)
+      expect($redis.get(PagesController::NUMBER_OF_SENTENCES)).to eq((example_json1.size.floor(-5) * 10).to_s)
     end
 
     it 'should set the NUMBER_OF_STUDENTS redis value' do
       subject.perform
-      expect($redis.get(PagesController::NUMBER_OF_STUDENTS)).to eq(@number_of_students.to_s)
+      expect($redis.get(PagesController::NUMBER_OF_STUDENTS)).to eq((example_json1.count('DISTINCT(user_id)').floor(-5)).to_s)
     end
 
     it 'should set the NUMBER_OF_TEACHERS redis value' do
       subject.perform
-      expect($redis.get(PagesController::NUMBER_OF_TEACHERS)).to eq(@number_of_teachers.to_s)
+      expect($redis.get(PagesController::NUMBER_OF_TEACHERS)).to eq("100")
```

Comment:
Much better to expect values to equal constants, but these current numbers are "magic".  I can't read this expectation and understand WHY this should be `"100"`.  Especially if I look above and see that `teachers` is a list of 150 teachers.  Why isn't this value 150?  I think you want this to be explicit in some way.  Either expecting `to eq(teachers.length.to_s)` or doing something like
```ruby
let(:num_teachers) { 150 }
let(:teachers) { create_list(:teacher, num_teachers) }

expect(...).to eq(num_teachers.to_s)
```
Same note on the two constant strings below on 52 and 57.